### PR TITLE
Changes Bookmark action to be a nav bar item

### LINKED ISSUE #872: Change Bookmark action to be a nav bar item
- Free up alert action space
- Simpler, more intuitive UI

I'll prepare some assets and PR them if someone wants to hack at this. I think this is the last big thing I want to do before launching 1.12.0!

@rizwankce @heshamsalman @Sherlouk @BasThomas for thoughts

### PR DESCRIPTION
**Pull request by rizwankce**
Closes #872

### CODE REVIEW

File: Classes/Repository/RepositoryViewController.swift [hunk 1]
```diff
@@ -81,6 +85,50 @@ NewIssueTableViewControllerDelegate {
...
+        }
+
+        if !store.contains(bookmark) {
+            store.add(bookmark)
+            sender.image = UIImage(named: "nav-bookmark-selected")
+        } else {
+            store.remove(bookmark)
+            sender.image = UIImage(named: "nav-bookmark")
+        }
+        Haptic.triggerNotification(.success)
```
Comment (rnystrom) [inline Classes/Repository/RepositoryViewController.swift]: A lot of this toggling code looks identical to the stuff in Issues. Think you could make a new file with a shared function both call sites can use? Similar to what I did when this was an alert.

Comment (rizwankce) [reply Classes/Repository/RepositoryViewController.swift]: Fair point 👍  Let me try!

**Comment by rizwankce** [comment]
@rnystrom Made some changes for shared toggle bookmark.

**Comment by heshamsalman** [comment]
The new bookmark nav bar item is _so sick_. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**